package/kedro_viz/api/rest/router.py

```diff
@@ -63,9 +63,7 @@ async def deploy_kedro_viz(input_values: S3DeployerConfiguration):
         )
     except Exception as exc:  # pragma: no cover
         logger.exception("Deploying Kedro Viz failed: %s ", exc)
-        return JSONResponse(
-            status_code=500, content={"message": "Failed to deploy Kedro Viz"}
-        )
+        return JSONResponse(status_code=500, content={"message": f"{exc}"})
```

Comment:
Not sure if we want to return the exception to the user (will appear in the developer tools even if handled on FE).

Reply:
I was thinking more on this. isn't it better to return something on the front-end on why the deploy failed for instance : S3 Credentials incorrect or S3FS not installed etc so the user knows how to fix things. 
presently, we don't do this on Kedro-viz, the user is expected to see the error on the terminal. So i am not sure how we want to tackle it here.

Reply:
it makes sense to return the exception to FE if we want FE to show the users exactly what is going wrong. So, I feel a better approach would be to show the users an error box on viz something like Deploy failed and an expandable div showing the exception details. I am not sure how a non-developer using shareable viz can sort this out (either in the UI or going through the dev tools).